### PR TITLE
ART-10668 Replace pip with uv when setting up art-tools venv

### PR DESCRIPTION
**Pull request by locriandev**
Use `uv` to set up a virtual environment for art-tools.

Requires https://github.com/openshift-eng/art-tools/pull/918

Test build: https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/hack/job/dpaolell/job/check-bugs/23/console

### CODE REVIEW

File: pipeline-scripts/buildlib.groovy [hunk 1]
```diff
@@ -173,16 +161,27 @@ def setup_venv() {
             throw err
         }
     }
-
     commonlib.shell(script: "rm -rf art-tools;  git clone -b ${branch_name} https://github.com/${owner}/art-tools.git")
-    commonlib.shell(script: "cd art-tools && ./install.sh")
```
Comment (thegreyd) [inline pipeline-scripts/buildlib.groovy:178]: I would like us to keep using install.sh, it's a good way to keep localdev, aos-cd-jobs init and gh workflow in sync

Comment (locriandev) [reply pipeline-scripts/buildlib.groovy:178]: Done

File: pipeline-scripts/buildlib.groovy [hunk 2]
```diff
@@ -173,16 +161,27 @@ def setup_venv() {
...
+    // Preparing venv for ART tools
+    // The following commands will run automatically every time one of our jobs
+    // loads buildlib (ideally, once per pipeline)
+    withEnv(['UV_LINK_MODE=symlink', 'PATH+MYCARGO=~/.cargo/bin']) {
+        VIRTUAL_ENV = "${env.WORKSPACE}/art-venv"
+        commonlib.shell(script: """
+            rm -rf ${VIRTUAL_ENV}
+            uv venv --python 3.11 --system-site-packages ${VIRTUAL_ENV}
+            source ${VIRTUAL_ENV}/bin/activate
+            uv pip install --upgrade pip setuptools
```
Comment (Ximinhan) [inline pipeline-scripts/buildlib.groovy:175]: curious if we can use --seed arg for uv venv

File: pipeline-scripts/buildlib.groovy [hunk 3]
```diff
@@ -173,16 +161,27 @@ def setup_venv() {
...
+            cd art-tools
+            ./install.sh
+            echo "Installed art-tools:"
+            pip list | grep 'doozer\\|elliott\\|pyartcd'
```
[the remark this answers is not in the document]
Comment (locriandev) [reply pipeline-scripts/buildlib.groovy]: right, updated

File: pipeline-scripts/buildlib.groovy (hunk 2 above)
Comment (locriandev) [reply pipeline-scripts/buildlib.groovy]: Looks like installing setuptools is not needed. I have removed this line from the script

**Pull request merged**